### ISSUE TITLE
Undefined symbol when implementing example code in XCode project

### ISSUE THREAD

**Issue by ariestav** [etr/libhttpserver#274]
Hello,

I'm trying to integrate this into a C++ library that I'm writing for a main host application.  I have included the .hpp header file and the compiler seems to know that various "symbols" exist.  During the linking phase of building, I am getting a series of "Undefined symbol:" errors.

When I look at the actual linker output it shows:

```
Undefined symbols for architecture arm64:
  "httpserver::resource_init(std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, bool, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, bool> > >*)", referenced from:
      httpserver::http_resource::http_resource() in 
...
```

Note that I'm building using XCode on an M1 (arm64) machine from Apple.  After running the `sudo make install` everything seems fine, but am curious if the library will link on an M1.

Thanks!

**Comment by etr** [comment]
Hey,

Thanks for reaching out. I suspect you don't have the symbols cached - in linux, you'd pre-cache this running ldconfig.

On mac, you might want to run something like

    sudo update_dyld_shared_cache

This should cache the symbols for you albeit you might have to reboot.

**Comment by ariestav** [comment]
Thank you for the suggestion.  I have tried that, but when I enter it in the command outputs:  "this tool is deprecated". Even after it reported that, I restarted and still the XCode compiler does not seem to have an issue locating the symbols while dev'ing.  It's only during Link phase.  I'm not sure why this is the case.  Any ideas?

**Comment by ariestav** [comment]
Also, not sure if it helps to understand the issue but I'm using libc++ not libstdc++ 

I'm not sure if that has any bearing, but any ideas are appreciated.

**Comment by LeSpocky** [comment]
Don't know XCode, but did you actually tell your project to link httpserver?  For compiling you need to set include paths, which is not necessary if those headers are installed in default locations like `/usr/include` or `/usr/local/include`.  (See GCC docs on [Options for Directory Search](https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html))

For linking you need to pass `-l` or `-Wl` or something like that (See GCC docs on [Options for Linking](https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html#Link-Options)).

Maybe you can have a look on how XCode calls GCC and what include and link options are actually passed?

.oO( Is XCode even using GCC, or some other compiler? 🤔 )

**Comment by ariestav** [comment]
@LeSpocky @etr Thank you for your time and help.  It turns out that I had linked the headers, but not the libraries.  In this way XCode was able to "know" about the symbols in the IDE, but during compile / linking time it didn't know because the libs were not included in XCode's set of search paths.  I've got it to build, and can get the server started but have run into another issue.  I'll create a new issue for that after searching to see if others have encountered the same issue — it's related to multithreading.
